core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionVerificationService.js

```diff
@@ -19,22 +19,29 @@
 oppia.factory('SolutionVerificationService', [
   '$injector', 'stateInteractionIdService', 'explorationContextService',
   'editorContextService', 'angularNameService', 'AnswerClassificationService',
+  'explorationStatesService',
   function(
     $injector, stateInteractionIdService, explorationContextService,
-    editorContextService, angularNameService, AnswerClassificationService) {
+    editorContextService, angularNameService, AnswerClassificationService,
+    explorationStatesService) {
     return {
       verifySolution: function(
         explorationId, state, correctAnswer, successCallback, errorCallback) {
         var interactionId = stateInteractionIdService.savedMemento;
         var rulesServiceName = (
           angularNameService.getNameOfInteractionRulesService(interactionId));
         var rulesService = $injector.get(rulesServiceName);
+        var validityBeforeVerification = (
+          explorationStatesService.isSolutionValid(
+            editorContextService.getActiveStateName()));
         AnswerClassificationService.getMatchingClassificationResult(
           explorationId, state, correctAnswer, true, rulesService
         ).then(function(result) {
           if (
             editorContextService.getActiveStateName() !== result.outcome.dest) {
-            successCallback();
+            if (!validityBeforeVerification) {
```

Comment:
Since the actual verification happens here, I'm not able to send the validity before verification as an argument.